core-services/00-pseudofs.sh

```diff
@@ -10,6 +10,10 @@ mountpoint -q /dev/pts || mount -o mode=0620,gid=5,nosuid,noexec -n -t devpts de
 mountpoint -q /dev/shm || mount -o mode=1777,nosuid,nodev -n -t tmpfs shm /dev/shm
 mountpoint -q /sys/kernel/security || mount -n -t securityfs securityfs /sys/kernel/security
 
+if [ -d /sys/firmware/efi/efivars ]; then
+    mountpoint -q /sys/firmware/efi/efivars || mount -o nosuid,noexec,nodev -t efivarfs efivarfs /sys/firmware/efi/efivars
+fi
```

Comment:
Whelp, please disregard!

Reply:
From `drivers/firmware/efi/efi.c`:

```c
        /* and the standard mountpoint for efivarfs */
        error = sysfs_create_mount_point(efi_kobj, "efivars");
        if (error) {
                pr_err("efivars: Subsystem registration failed.\n");
                goto err_remove_group;
        }
```

So that mount point is created by efisubsys_init only if the system is booted with EFI and EFI services are available, etc. What is "sysfs_create_mount_point"? Sounds kind of fancy, but actually:

```c
/**
 * sysfs_create_mount_point - create an always empty directory
 * @parent_kobj:  kobject that will contain this always empty directory
 * @name: The name of the always empty directory to add
 */
int sysfs_create_mount_point(struct kobject *parent_kobj, const char *name)
{
        struct kernfs_node *kn, *parent = parent_kobj->sd;

        kn = kernfs_create_empty_dir(parent, name);
        if (IS_ERR(kn)) {
                if (PTR_ERR(kn) == -EEXIST)
                        sysfs_warn_dup(parent, name);
                return PTR_ERR(kn);
        }

        return 0;
}
```

It just makes an empty directory. So I think the `[ -d` check is actually the correct thing to do.